sycl/test-e2e/USM/align.cpp

```diff
@@ -35,7 +29,10 @@ template <typename T> void testAlign(sycl::queue &q, unsigned align) {
     return aligned_alloc_host(align, N, args...);
   };
   auto AShared = [&](size_t align, auto... args) {
-    return aligned_alloc_shared(align, N, args...);
+    if (dev.has(aspect::usm_shared_allocations)) {
+      return aligned_alloc_shared(align, N, args...);
+    }
+    return (void *)nullptr;
```

Comment:
Compilation was failing on my machine without the cast. I will apply the change however and see if I get the same error.

Reply:
@ldrumm I've reverted back to using the casts because type deduction for the lambda fails otherwise since the types deduced for `nullptr` and the `aligned_alloc_shared` function call are different.

Reply:
Oh right. Thanks for checking. C++ is wild